src/NLog/Layouts/Typed/Layout.cs

```diff
@@ -0,0 +1,315 @@
+//   this list of conditions and the following disclaimer. 
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+using System;
+using System.ComponentModel;
+using System.Globalization;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using JetBrains.Annotations;
+using NLog.Common;
+using NLog.Config;
+using NLog.Internal;
+
+namespace NLog.Layouts
+{
+    /// <summary>
+    /// Layout with a simple value (e.g. int) or a layout which results in a simple value (e.g. ${counter})
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class Layout<T> : Layout, IRenderable<T>
+    {
+        private static Type _type;
+        private static readonly string _typeNamed;
+        private readonly Layout _layout;
+        private readonly T _value;
+        private readonly bool _fixedValue;
+
+        /// <inheritdoc />
+        public Layout(T value)
+        {
+            _value = value;
+            _fixedValue = true;
+            _layout = null;
+        }
+
+        /// <inheritdoc />
+        public Layout(Layout layout)
+        {
+            _fixedValue = TryGetFixedValue(layout, out _value);
+            _layout = layout;
+        }
+
+        static Layout()
+        {
+            var type = typeof(T);
+            if (IsNullable(type))
+            {
+                var arg = type.GetGenericArguments()[0];
+
+#if !NETSTANDARD1_3 //todo fix
+
+                _type = arg;
+
+#endif
+            }
+            else
+            {
+                _type = type;
+            }
+
+            _typeNamed = type.Name;
+        }
+
+        private static bool IsNullable(Type propertyType)
+        {
+            return propertyType.IsGenericType() && propertyType.GetGenericTypeDefinition() == typeof(Nullable<>);
```

Comment:
Why not `Nullable.GetUnderlyingType(propertyType) ?? propertyType;` ?

Reply:
That's indeed better!

I made this first with TypeInfo, but switched to Type for .NET Standard 1.3

Reply:
Fixed